InteractiveHtmlBom/web/render.js

```diff
@@ -459,19 +472,25 @@ function handleMouseClick(e, layerdict) {
   }
 }
 
-function handleMouseUp(e, layerdict) {
+function handlePointerLeave(e, layerdict) {
+  e.preventDefault();
+  e.stopPropagation();
+
+  if (!redrawOnDrag) {
+    redrawCanvas(layerdict);
+  }
+
+  delete layerdict.pointerStates[e.pointerId];
+}
+
+function handlePointerUp(e, layerdict) {
   e.preventDefault();
   e.stopPropagation();
-  if (e.which == 1 &&
-    layerdict.transform.mousedown &&
-    layerdict.transform.mousedownx == e.offsetX &&
-    layerdict.transform.mousedowny == e.offsetY) {
+
+  if (e.button == 0 && layerdict.pointerStates.hasOwnProperty(e.pointerId) && layerdict.pointerStates[e.pointerId].distanceTravelled < 10) {
     // This is just a click
     handleMouseClick(e, layerdict);
-    layerdict.transform.mousedown = false;
-    return;
-  }
-  if (e.which == 3) {
+  } else if (e.button == 2) {
```

Comment:
What is equivalent of right click with touch screens?
Can we have two finger tap act as reset (similar to how right click is emulated on laptop touchpads)?

Reply:
Hmm, if you want? I think the code to detect this while avoiding false detections could get slightly messy, and it's generally just a single quick pinch gesture to get back out, albiet not perfectly centered

Reply:
I think this is worth doing.
Suggestion on detection algorithm: since we track distanceTraveled for each pointer don't throw it away when two pointers are detected. Instead remember if one was released with < 10 distance and then if later second one was also released with < 10 distance traveled trigger reset. As additional signal we can measure time of pointer press/release and not trigger reset if it's more than, say 500ms for each.